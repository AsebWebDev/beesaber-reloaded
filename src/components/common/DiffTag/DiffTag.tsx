import React from 'react';
import styled from 'styled-components';

import type { Difficulty, DifficultyName } from '@/sharedTypes/UserScores';

const Tag = styled.span`
  margin: 1px 2px 1px 2px;
  max-height: 20px;
  max-width: fit-content;
  min-width: 50px;
`;

type BadgeColor =
  | 'danger'
  | 'default'
  | 'primary'
  | 'secondary'
  | 'success'
  | 'warning';

type Props = {
  difficulty: Difficulty;
};

const DiffTag = ({ difficulty }: Props): JSX.Element => {
  let badgeColor: BadgeColor;
  let badgeText: DifficultyName | 'Oops';

  switch (difficulty) {
    case 1:
      badgeColor = 'success';
      badgeText = 'Easy';
      break;
    case 3:
      badgeColor = 'primary';
      badgeText = 'Normal';
      break;
    case 5:
      badgeColor = 'warning';
      badgeText = 'Hard';
      break;
    case 7:
      badgeColor = 'secondary';
      badgeText = 'Expert';
      break;
    case 9:
      badgeColor = 'danger';
      badgeText = 'Expert+';
      break;
    default:
      badgeColor = 'default';
      badgeText = 'Oops';
      break;
  }

  return <Tag className={`badge badge-${badgeColor}`}>{badgeText}</Tag>;
};

export default DiffTag;
