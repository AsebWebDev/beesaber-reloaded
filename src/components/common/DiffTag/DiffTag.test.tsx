import { render } from '@testing-library/react';

import DiffTag from './DiffTag';

import type { Difficulty } from '@/../sharedTypes/UserScores';

describe('components/common/DiffTag', () => {
  it.each([1, 3, 5, 7, 9] as Difficulty[])(
    'should match snapshot when diffiultiy is %s',
    (difficulty) => {
      const { container } = render(<DiffTag difficulty={difficulty} />);

      expect(container.firstChild).toMatchSnapshot();
    }
  );
});
