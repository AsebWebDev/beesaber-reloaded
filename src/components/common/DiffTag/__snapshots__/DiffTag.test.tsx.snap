// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/common/DiffTag should match snapshot when diffiultiy is 1 1`] = `
.c0 {
  margin: 1px 2px 1px 2px;
  max-height: 20px;
  max-width: -webkit-fit-content;
  max-width: -moz-fit-content;
  max-width: fit-content;
  min-width: 50px;
}

<span
  class="c0 badge badge-success"
>
  Easy
</span>
`;

exports[`components/common/DiffTag should match snapshot when diffiultiy is 3 1`] = `
.c0 {
  margin: 1px 2px 1px 2px;
  max-height: 20px;
  max-width: -webkit-fit-content;
  max-width: -moz-fit-content;
  max-width: fit-content;
  min-width: 50px;
}

<span
  class="c0 badge badge-primary"
>
  Normal
</span>
`;

exports[`components/common/DiffTag should match snapshot when diffiultiy is 5 1`] = `
.c0 {
  margin: 1px 2px 1px 2px;
  max-height: 20px;
  max-width: -webkit-fit-content;
  max-width: -moz-fit-content;
  max-width: fit-content;
  min-width: 50px;
}

<span
  class="c0 badge badge-warning"
>
  Hard
</span>
`;

exports[`components/common/DiffTag should match snapshot when diffiultiy is 7 1`] = `
.c0 {
  margin: 1px 2px 1px 2px;
  max-height: 20px;
  max-width: -webkit-fit-content;
  max-width: -moz-fit-content;
  max-width: fit-content;
  min-width: 50px;
}

<span
  class="c0 badge badge-secondary"
>
  Expert
</span>
`;

exports[`components/common/DiffTag should match snapshot when diffiultiy is 9 1`] = `
.c0 {
  margin: 1px 2px 1px 2px;
  max-height: 20px;
  max-width: -webkit-fit-content;
  max-width: -moz-fit-content;
  max-width: fit-content;
  min-width: 50px;
}

<span
  class="c0 badge badge-danger"
>
  Expert+
</span>
`;
